src/graph.rs

```diff
@@ -354,6 +302,78 @@ impl From<String> for Value {
     }
 }
 
+impl fmt::Display for Value {
```

Comment:
Actually, as a separate question from the comments below about the implementation of this trait — do we _need_ to implement `Display` for `Value`?  Are there places where we don't have access to a `Graph`, and therefore can't call `value.display_with` instead?

Reply:
In principle we always ahve the graph. But, there are two places where I couldn't get it to work because of borrowing:
- https://github.com/tree-sitter/tree-sitter-graph/blob/improve-error-messages/src/execution.rs#L350-L351
- https://github.com/tree-sitter/tree-sitter-graph/blob/improve-error-messages/src/execution.rs#L370-L371
If we can fix that, we could drop Display impl altogether.

Reply:
Ah yes!  Just played around with it locally, and this pattern works:

``` rust
        if let Err(_) = exec.graph[source].add_edge(sink) {
            return Err(ExecutionError::DuplicateEdge(format!(
                "({} -> {}) in {}",
                source.display_with(exec.graph), // cannot display_with(exec.graph) because of borrow
                sink.display_with(exec.graph), // cannot display_with(exec.graph) because of borrow
                self.display_with(ctx)
            )));
        }
```

That way you still have a mutable borrow of `exec` in the conditional, but it's discharged before control passes into the body.

Reply:
``` rust
if exec.graph[source].add_edge(sink).is_err() {
```

would also work for the conditional if you prefer that spelling.

Reply:
I think I found a way around this, by matching instead of calling `.map_err` in the chain.

Reply:
I managed. I have removed the Display impl in favor of DisplayWithGraph. If this is too quickly, let me know and I revert it easily.